lectures/zksnark.tex

```diff
@@ -0,0 +1,271 @@
+        \centering
+        % Boolean AND and OR gates
+        \begin{tabular}{cc}
+            \begin{tikzpicture}
+                % Nodes
+                \node[var] (a) at (0, -1.5) {$a$};
+                \node[var] (b) at (2, -1.5) {$b$};
+                \node[gate] (and) at (1, 0) {and};
+                \node[var] (c) at (1, 1.5) {$c$};
+
+                % Arrows
+                \draw[arrow] (a) -- (and);
+                \draw[arrow] (b) -- (and);
+                \draw[arrow, blue] (and) -- (c);
+            \end{tikzpicture}
+            &
+            \begin{tikzpicture}
+                % Nodes
+                \node[var] (a) at (0, -1.5) {$a$};
+                \node[var] (b) at (2, -1.5) {$b$};
+                \node[gate] (or) at (1, 0) {or};
+                \node[var] (c) at (1, 1.5) {$c$};
+
+                % Arrows
+                \draw[arrow] (a) -- (or);
+                \draw[arrow] (b) -- (or);
+                \draw[arrow, blue] (or) -- (c);
+            \end{tikzpicture}
+        \end{tabular}
+        \caption{Boolean AND and OR Gates}
+    \end{minipage}
+    \hspace{0.05\textwidth} % Space between figures
+    \begin{minipage}{0.46\textwidth}
+        \centering
+        % Addition and Multiplication gates
+        \begin{tabular}{cc}
+            \begin{tikzpicture}
+                % Nodes
+                \node[var] (a) at (0, -1.5) {$a$};
+                \node[var] (b) at (2, -1.5) {$b$};
+                \node[gate] (add) at (1, 0) {$+$};
+                \node[var] (c) at (1, 1.5) {$c$};
+
+                % Arrows
+                \draw[arrow] (a) -- (add);
+                \draw[arrow] (b) -- (add);
+                \draw[arrow, blue] (add) -- (c);
+            \end{tikzpicture}
+            &
+            \begin{tikzpicture}
+                % Nodes
+                \node[var] (a) at (0, -1.5) {$a$};
+                \node[var] (b) at (2, -1.5) {$b$};
+                \node[gate] (mul) at (1, 0) {$\times$};
+                \node[var] (c) at (1, 1.5) {$c$};
+
+                % Arrows
+                \draw[arrow] (a) -- (mul);
+                \draw[arrow] (b) -- (mul);
+                \draw[arrow, blue] (mul) -- (c);
+            \end{tikzpicture}
+        \end{tabular}
+        \caption{Addition and Multiplication Gates}
+    \end{minipage}
+
+    \vspace*{1em}
+\end{figure}
+
+\begin{wraptable}{r}{0.4\textwidth}
+    \centering
+    \vspace{-1em}
+    \begin{tabular}{|c|c|c|}
+        \hline
+        \textbf{A} & \textbf{B} & \textbf{A AND B} \\
+        \hline
+        0 & 0 & 0 \\
+        \hline
+        0 & 1 & 0 \\
+        \hline
+        1 & 0 & 0 \\
+        \hline
+        1 & 1 & 1 \\
+        \hline
+    \end{tabular}
+    \caption{AND Gate Truth Table}
+    \label{tab:and-truth-table}
+    \vspace{1em}
+\end{wraptable}
+
+% --- Finish Writing diagrams ---
+
+The \textit{AND Gate Truth Table~\ref{tab:and-truth-table}} shows us the results we receive if 
```

Comment:
Instead of `Table~\ref{...}` you can instead use `Cref{...}` using the [`cleveref`](https://ctan.org/pkg/cleveref?lang=en) package